cmd/extensions/oidc-policy/internal/controller/oidcpolicy_reconciler.go

```diff
@@ -117,14 +119,14 @@ func (r *OIDCPolicyReconciler) Reconcile(ctx context.Context, request reconcile.
 
 	r.Logger.V(1).Info("Resolving ingress gateway info", "ingressGatewayData", ingressGatewayData)
 
-	specErr := r.reconcileSpec(ctx, oidcPolicy, &ingressGatewayData)
-
-	statusResult, statusErr := r.reconcileStatus(ctx, oidcPolicy, specErr)
+	mainAuthPol, specErr := r.reconcileSpec(ctx, oidcPolicy, &ingressGatewayData)
 
 	if specErr != nil {
 		return reconcile.Result{}, specErr
 	}
```

Comment:
if there's an error reconciling spec shouldn't status reflect it?

Reply:
We don't have to do this here, but I wonder if in future instead we should wrap the user defined reconcile function in the extensions framework `controllers.go` to set a default (overrideable) status in this case

Reply:
Ah can be addressed likely in https://github.com/Kuadrant/kuadrant-operator/issues/1484